webrender/src/frame.rs

```diff
@@ -296,6 +298,40 @@ impl Frame {
     }
 
     /// Returns true if any layers actually changed position or false otherwise.
+    pub fn scroll_layers(&mut self,
+                         origin: Point2D<f32>,
+                         pipeline_id: PipelineId,
+                         scroll_root_id: ServoScrollRootId)
+                          -> bool {
+        let origin = LayerPoint::new(origin.x.max(0.0), origin.y.max(0.0));
+
+        let mut scrolled_a_layer = false;
+        let mut found_layer = false;
+        for (layer_id, layer) in self.layers.iter_mut() {
+            if layer_id.pipeline_id != pipeline_id {
+                continue;
+            }
+
+            match layer_id.info {
+                ScrollLayerInfo::Scrollable(_, id) if id != scroll_root_id => continue,
+                ScrollLayerInfo::Fixed => continue,
+                _ => {}
+            }
+
+            found_layer = true;
+            scrolled_a_layer = layer.set_scroll_origin(&origin);
```

Comment:
probably you want this to be `scrolled_a_layer |= layer.set_scroll_origin(&origin)`, right?

Otherwise (if we can just scroll one layer) we might as well break from the loop when that happens, right?